packages/manager/apps/pci-rancher/src/pages/create/index.tsx

```diff
@@ -35,7 +35,9 @@ export default function Create() {
 
   const { data: project } = usePciProject();
 
-  const { data: ranchers, refetch: refetchRancherList } = useRanchers();
+  const { data: ranchers, refetch: refetchRancherList } = useRanchers({
+    shouldDisableRefetch: true,
+  });
```

Comment:
**nitpick:** We should make an HTTP call just to decide where should we take user on cancel click. This can be tackled in 2 ways,
1. In `CreateRancher` component `onCancelClick` function,  you can do `navigate(-1)`
2. Otherwise, you must have check in the Rancher listing component and in the `useEffect({}, [])` if the Rancher list is empty, redirect user to onboarding page.

Reply:
don't know if that help but navigate('..') would navigate to the parent route no matter where the user comes from